### PR TITLE
Port EEM and new services experience docs to stateful

### PR DESCRIPTION
**Pull request by dedemorton**
Ports #4062 to serverless.  Also makes some minor tweaks to serverless docs to mirror structural changes needed for stateful docs.

Previews:
* [New services experience](https://observability-docs_bk_4113.docs-preview.app.elstc.co/guide/en/observability/master/new-experience-services.html) 
* [Elastic Entity Model](https://observability-docs_bk_4113.docs-preview.app.elstc.co/guide/en/observability/master/elastic-entity-model.html)

TODO:

- [x] Add topics to the correct place in the nav after #3723 is merged.
- [x] Finish porting all changes
- [x] Update link when topic about defining service.name has been ported to stateful (requires https://github.com/elastic/observability-docs/pull/4116)

### CODE REVIEW

File: docs/en/observability/elastic-entity-model.asciidoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+In the context of Elastic Observability,
+an _entity_ is an object of interest that can be associated with produced telemetry and identified as unique.
+Note that this definition is intentionally closely aligned to the work of the https://github.com/open-telemetry/oteps/blob/main/text/entities/0256-entities-data-model.md#data-model[OpenTelemetry Entities SIG].
+Examples of entities include (but are not limited to) services, hosts, and containers.
+
+The concept of an entity is important as a means to unify observability signals based on the underlying entity that the signals describe.
+
+[NOTE]
+====
+* The Elastic Entity Model currently supports the <<new-experience-services,new service inventory experience>> limited to service-based entities (as identified by `service.name`) located in data identified by `logs-*` and `trace-*` index patterns
```
Comment (klacabane) [inline docs/en/observability/elastic-entity-model.asciidoc:21]: data identified by `logs-*` and `filebeat*`. There's also `metrics-apm.service_transaction.1m*` but it is used for metrics aggregation, although services available in the metrics but not in logs will still be extracted (see [relevant code](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/entity_manager/server/lib/entities/built_in/services.ts#L30))

File: docs/en/observability/elastic-entity-model.asciidoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+|====
+|**Index privileges**
+|names: [`.entities*`], privileges: [`create_index`, `index`, `create_doc`, `auto_configure`, `read`]
+
+names: [`logs-*`, `filebeat*`, `metrics-*`, `metricbeat*`, `traces-*`, `.entities*`], privileges: [`read`, `view_index_metadata`],
+|**Cluster privileges**
+|`manage_transform`, `manage_ingest_pipelines`, `manage_index_templates`
+
+|**Application privileges**
+|application: `kibana-.kibana`, privileges: [`saved_object:entity-definition/*`, `saved_object:entity-discovery-api-key/*`], resources: [*]
```
Comment (klacabane) [inline docs/en/observability/elastic-entity-model.asciidoc:43]: LGTM

File: docs/en/observability/elastic-entity-model.asciidoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+From the Dev Console, run the command: `DELETE kbn:/internal/entities/managed/enablement`
+
+The following {es} privileges are required to delete {es} resources:
+
+[cols="1,1"]
+|====
+|**Index privileges**
+|names: [`.entities*`], privileges: [`delete_index`]
+
+names: [`logs-*`, `filebeat*`, `metrics-*`, `metricbeat*`, `traces-*`, `.entities*`], privileges: [`read`, `view_index_metadata`],
```
Comment (klacabane) [inline docs/en/observability/elastic-entity-model.asciidoc:60]: this is not required to disable eem

File: docs/en/serverless/elastic-entity-model.mdx [hunk 4]
```diff
@@ -22,19 +22,20 @@ Examples of entities include (but are not limited to) services, hosts, and conta
 
 The concept of an entity is important as a means to unify observability signals based on the underlying entity that the signals describe.
 
-## Enable or disable the Elastic Entity Model
+<DocCallOut title="Notes">
+    - The Elastic Entity Model currently supports the <DocLink slug="/serverless/observability/new-experience-services">new service inventory experience</DocLink> limited to service-based entities (as identified by `service.name`) located in data identified by `logs-*` and `trace-*` index patterns
```
Comment (klacabane) [inline docs/en/serverless/elastic-entity-model.mdx:26]: same as above, we don't read from `trace-*` but `filebeat*`

File: docs/en/observability/elastic-entity-model.asciidoc (hunk 1 above)
Comment (dedemorton) [inline docs/en/observability/elastic-entity-model.asciidoc:21]:
```suggestion
* The Elastic Entity Model currently supports the <<new-experience-services,new service inventory experience>> limited to service-based entities (as identified by `service.name`) located in data identified by `logs-*` and `filebeat*` index patterns
```

File: docs/en/observability/elastic-entity-model.asciidoc (hunk 3 above)
Comment (dedemorton) [inline docs/en/observability/elastic-entity-model.asciidoc:60]:
```suggestion
```

Comment (dedemorton) [reply docs/en/observability/elastic-entity-model.asciidoc:60]: Sorry, copy/paste error. I should have double-checked.

File: docs/en/serverless/elastic-entity-model.mdx (hunk 4 above)
Comment (dedemorton) [inline docs/en/serverless/elastic-entity-model.mdx:26]:
```suggestion
    - The Elastic Entity Model currently supports the <DocLink slug="/serverless/observability/new-experience-services">new service inventory experience</DocLink> limited to service-based entities (as identified by `service.name`) located in data identified by `logs-*` and `filebeat*` index patterns
```

**Comment by dedemorton** [comment]
@klacabane I've applied your changes, and this PR should be ready for final approval (assuming the doc builds pass). Thanks for your input!

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-3jpfgzwcj-elastic-dev.vercel.app

* Built with commit c63509a6650ce0ff39f04a8f37a950f2709b8743

> Issues? Visit #next-docs in Slack

File: docs/en/observability/elastic-entity-model.asciidoc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+the Elastic Entity Model is enabled when you turn on the entity-centric service inventory described in <<new-experience-services>>.
+
+The following {es} privileges are required:
+
+[cols="1,1"]
+|====
+|**Index privileges**
+|names: [`.entities*`], privileges: [`create_index`, `index`, `create_doc`, `auto_configure`, `read`]
+
+names: [`logs-*`, `filebeat*`, `metrics-*`, `metricbeat*`, `traces-*`, `.entities*`], privileges: [`read`, `view_index_metadata`],
```
Comment (mdbirnstiehl) [inline docs/en/observability/elastic-entity-model.asciidoc:38]:
```suggestion
names: [`logs-*`, `filebeat*`, `metrics-*`, `metricbeat*`, `traces-*`, `.entities*`], privileges: [`read`, `view_index_metadata`]
```
Looks like maybe an extra comma here?

**Review by mdbirnstiehl — COMMENTED** [review]
One small comment, otherwise LGTM!

File: docs/en/observability/elastic-entity-model.asciidoc (hunk 5 above)
Comment (dedemorton) [reply docs/en/observability/elastic-entity-model.asciidoc:38]: No, I think you're right. It was in the source doc, too, but it seems extraneous.

File: docs/en/observability/elastic-entity-model.asciidoc [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+The following {es} privileges are required:
+
+[cols="1,1"]
+|====
+|**Index privileges**
+|names: [`.entities*`], privileges: [`create_index`, `index`, `create_doc`, `auto_configure`, `read`]
+
+names: [`logs-*`, `filebeat*`, `metrics-*`, `metricbeat*`, `traces-*`, `.entities*`], privileges: [`read`, `view_index_metadata`]
+|**Cluster privileges**
```
Comment (dedemorton) [inline docs/en/observability/elastic-entity-model.asciidoc:39]:
```suggestion
names: [`logs-*`, `filebeat*`, `metrics-*`, `metricbeat*`, `traces-*`, `.entities*`], privileges: [`read`, `view_index_metadata`]

|**Cluster privileges**
```

**Comment by dedemorton** [comment]
I'm going to merge this. If there are any additional changes required, I can follow up with another PR. Thanks everyone!

**Pull request merged**